DevelopGitHub/GitHubExtension/GitHubExtension.LocalizationTool/ViewModel/MainWindowViewModel.cs

```diff
@@ -0,0 +1,257 @@
+                }
+                return saveJsonCommand;
+            }
+        }
+
+        private void SaveJson(object param)
+        {
+            JsonHelper.RemoveEmptyRows();
+            foreach (Lang value in Enum.GetValues(typeof(Lang)))
+            {
+                File.WriteAllText(Translator.GetFileName(value), JsonHelper.GenerateJson(value));
+            }
+
+            ShowInformationMessageBox("Saved!");
+        }
+
+        private DelegateCommand openJsonCommand;
+
+        public ICommand OpenJsonCommand
+        {
+            get
+            {
+                if (openJsonCommand == null)
+                {
+                    openJsonCommand = new DelegateCommand(OpenJson);
+                }
+                return openJsonCommand;
+            }
+        }
+
+        private void OpenJson(object param)
+        {
+            TranslationData.Clear();
+            foreach (Lang value in Enum.GetValues(typeof(Lang)))
+            {
+                JsonHelper.ReadJsonFromFile(value);
+            }
+
+            JsonHelper.RemoveEmptyRows();
+            ShowInformationMessageBox("Oppened!");
+        }
+
+        private DelegateCommand clearTranslationDataCommand;
+
+        public ICommand ClearTranslationDataCommand
+        {
+            get
+            {
+                if (clearTranslationDataCommand == null)
+                {
+                    clearTranslationDataCommand = new DelegateCommand(ClearTranslationData);
+                }
+                return clearTranslationDataCommand;
+            }
+        }
+
+        private void ClearTranslationData(object param)
+        {
+            TranslationData.Clear();
+        }
+
+        private DelegateCommand translatingButtonCommand;
+
+        public ICommand TranslatingButtonCommand
+        {
+            get
+            {
+                if (translatingButtonCommand == null)
+                {
+                    translatingButtonCommand = new DelegateCommand(TranslatingButton);
+                }
+                return translatingButtonCommand;
+            }
+        }
+
+        private async void TranslatingButton(object param)
+        {
+            TranslateButtonIsEnabled = false;
+            TranslateButtonContent = "Translating...";
+            await Translator.WebTranslate(SourceLanguageText, TargetLanguageText);
+            TranslateButtonIsEnabled = true;
+            TranslateButtonContent = "WebTranslate";
+        }
+
+        private DelegateCommand closeCommand;
+
+        public ICommand CloseCommand
+        {
+            get
+            {
+                if (closeCommand == null)
+                {
+                    closeCommand = new DelegateCommand(Close);
+                }
+                return closeCommand;
+            }
+        }
+
+        private void Close(object param)
+        {
+            Application.Current.MainWindow.Close();
```

Comment:
This can be extracted to the command and binded directly in XAML that you can reuse it.